source/grid/grid_tools_dof_handlers.cc

```diff
@@ -465,45 +465,47 @@ namespace GridTools
             endc = adjacent_cells.end();
           for (; cell != endc; ++cell)
             {
-              try
+              if ((*cell)->is_locally_owned())
```

Comment:
I am not quite sure why we should exclude ghost cells.

Reply:
I would say those points are found by other processors

Reply:
The first question I could not answer in a short search is whether we actually exclude artificial cells. That should be the bare minimum to request here because all mappings that read data from solution vectors would fail. Then the point is whether to include ghosts or not - depends on the context, there certainly are cases where one would like to exclude ghosts, but I am not sure if that is unconditionally true.

Reply:
Indeed the problem are the artificial cells. So, I would suggest:

```suggestion
              if((*cell)->is_artificial() == false)
````

Or, alternatively: we could amend `find_cells_adjacent_to_vertex` so that it does not return artificial cells.  Any preferences?

Reply:
The documentation specifies that the function may return ghosted cells, so we should probably adhere to this specification and use ``is_artificial() == false``-

Reply:
> there certainly are cases where one would like to exclude ghosts, but I am not sure if that is unconditionally true.

One problem with ignoring ghost cells is that due to tolerances, two processors at a cell boundary where you query might decide that the point is outside the own cell. If you ignore ghost cells, you might have no way to locate a cell.